app/root.tsx

```diff
@@ -61,18 +97,27 @@ export default function Document() {
                 <Links />
             </head>
             <body>
-                <MantineProvider
-                    withGlobalStyles
-                    withNormalizeCSS
-                    emotionCache={emotionCache}
-                    theme={theme}
+                <ColorSchemeProvider
+                    colorScheme={colorScheme}
+                    toggleColorScheme={toggleColorScheme}
                 >
-                    <NotificationsProvider>
-                        <AppShell padding='md' header={<Header />}>
-                            <Outlet />
-                        </AppShell>
-                    </NotificationsProvider>
-                </MantineProvider>
+                    <MantineProvider
+                        key={colorScheme}
```

Comment:
question: Do we have to remount the provider for this to work? Is passing the new theme prop not enough? If not, why doesn't the `ColorSchemeProvider` need a key prop too?